config/babylon_query_config.go

```diff
@@ -0,0 +1,26 @@
+package config
+
+import (
+	"errors"
+	"time"
+)
+
+// BabylonConfig defines configuration for the Babylon query client
+type BabylonQueryConfig struct {
+	RPCAddr string        `mapstructure:"rpc-addr"`
+	Timeout time.Duration `mapstructure:"timeout"`
+}
+
+func (cfg *BabylonQueryConfig) Validate() error {
+	if cfg.Timeout <= 0 {
+		return errors.New("cfg.Timeout must be positive")
+	}
```

Comment:
Maybe we can also validate the format of `RPCAddr` here?

Reply:
Good idea! done